educator/src/Dscp/Educator/Web/Educator/Queries.hs

```diff
@@ -248,3 +253,21 @@ educatorPostGrade subH grade = do
             , trIdx = TxInMempool
             , trSubmission = packPk subH
             }
+
+storeCertificateOlolo :: CertificateMeta -> PDFBody -> m ()
+storeCertificateOlolo _ _ = error "Implement"
+
+educatorAddCertificate
+    :: MonadEducatorWeb ctx m
+    => CertificateFullInfo -> m ()
+educatorAddCertificate cert = do
+    mblk <- transact $ createPrivateBlockFromCert cert
+    -- This private block will be added to the public chain on itself shortly
+
+    eAddr <- ourAddress @EducatorNode
+    let blk = mblk ?: error "What if there were no grades provided?"
```

Comment:
Good point. Diploma without grades is something absurd, so I guess we should actually make `cfiGrades :: NonEmpty CertificateGrade` (and also note in the endpoint description that empty list of grades will result in error).